subcommand/connect-init/command_test.go

```diff
@@ -209,29 +217,28 @@ func TestRun_ServicePollingWithACLsAndTLS(t *testing.T) {
 			}
 			// Run the command.
 			code := cmd.Run(flags)
-			if test.expErr != "" {
-				require.Equal(t, 1, code)
-				require.Contains(t, ui.ErrorWriter.String(), test.expErr)
-			} else {
-				require.Equal(t, 0, code, ui.ErrorWriter.String())
-
-				// Validate the ACL token was written.
-				tokenData, err := ioutil.ReadFile(tokenFile)
-				require.NoError(t, err)
-				require.NotEmpty(t, tokenData)
-
-				// Check that the token has the metadata with pod name and pod namespace.
-				consulClient, err = api.NewClient(&api.Config{Address: server.HTTPAddr, Token: string(tokenData)})
-				require.NoError(t, err)
-				token, _, err := consulClient.ACL().TokenReadSelf(nil)
-				require.NoError(t, err)
-				require.Equal(t, "token created via login: {\"pod\":\"default-ns/counting-pod\"}", token.Description)
-
-				// Validate contents of proxyFile.
-				data, err := ioutil.ReadFile(proxyFile)
-				require.NoError(t, err)
-				require.Contains(t, string(data), "counting-counting-sidecar-proxy")
+			if test.expCode != 0 {
+				require.Equal(t, test.expCode, code)
+				return
```

Comment:
I really like this fixture. The only suggestion I have is potentially using a `bool` for whether an error is expected or not. If it is expected, assert that the return code is 1, else assert that it is 0. That can make the if check on line 220 a little cleaner maybe? 
`if test.expError {}`

Reply:
Love it, great suggestion. When I was poking around at it I was thinking that expCode here kinda looked funky but couldnt think of a better way. thanks!